mantidproject/mantid
# Rework function browser in ConvFit tab of IDA GUI

**Description of work.**
This PR reworks the function browser in the ConvFit tab of the IDA GUI. The main objective of this rework is to allow Lorentzians to be combined with other functions defined within the tab.

This PR encountered an issue with the spinbox defined as part of `QtIntPropertyManager`. When this value is changed, it kicks off a timer which is used to simulate `mouse hold` events. However, when the slot triggers a processing stage which takes a "long" time, the timer event falsely goes off, which leads to the spinbox being incremented (meaning it will increment to its maximum value on the first step). Ideally we should be able to detatch that timer for particular use cases (such as this one). A similar issue can be found in the QT bug report https://bugreports.qt.io/browse/QTBUG-33128. To get round this issue for the time being we have offloaded the main processing to Qt threads, which allows us to continue processing the `QtEvents` (which ceases the spin box timer event). An issue to deal with this can be found in #29046



**To test:**

- Data to test each tab of the Indirect Data Analysis GUI can be found in #26631.
- Open the Indirect Data Analysis GUI
- Go to the ConvFit tab
- Load in data and a resolution file
- Add a single lorentzian using the new spinner
- Add a fit type
- Do a fit (this should all work)
- Play around with the GUI checking other functionality works e.g Switching between the full function view using the checkbox


This is part of a minor set of changes to the IDA GUI #28484




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
